aiida/manage/caching.py

```diff
@@ -163,54 +188,123 @@ def _reset_config():
+            _validate_identifier_pattern(identifier=identifier)
             _CONFIG[ConfigKeys.DISABLED.value].append(identifier)
-            try:
+            with suppress(ValueError):
                 _CONFIG[ConfigKeys.ENABLED.value].remove(identifier)
-            except ValueError:
-                pass
         yield
+
+
+def _match_wildcard(*, string, pattern):
+    """
+    Helper function to check whether a given name matches a pattern
+    which can contain '*' wildcards.
+    """
+    regexp = '.*'.join(re.escape(part) for part in pattern.split('*'))
+    return re.fullmatch(pattern=regexp, string=string) is not None
+
+
+def _validate_identifier_pattern(*, identifier):
+    """
+    The identifier (without wildcards) can have one of two forms:
+
+    1.  <group_name><ENTRY_POINT_STRING_SEPARATOR><tail>
+
+        where `group_name` is one of the keys in `entry_point_group_to_module_path_map`
+        and `tail` can be anything _except_ `ENTRY_POINT_STRING_SEPARATOR`.
+
+    2. a fully qualified Python name
+
+        this is a colon-separated string, where each part satisfies
+        `part.isidentifier() and not keyword.iskeyword(part)`
+
+    This function checks if an identifier _with_ wildcards can possibly
+    match one of these two forms. If it can not, a `ValueError` is raised.
+
+    :param identifier: Process type string, or a pattern with '*' wildcard that matches it.
+    :type identifier: str
+    """
+    common_error_msg = "Invalid identifier pattern '{}': ".format(identifier)
+    assert ENTRY_POINT_STRING_SEPARATOR not in '.*'  # The logic of this function depends on this
+    # Check if it can be an entry point string
+    if identifier.count(ENTRY_POINT_STRING_SEPARATOR) > 1:
+        raise ValueError(
+            common_error_msg +
+            "Can contain at most one entry point string separator '{}'".format(ENTRY_POINT_STRING_SEPARATOR)
+        )
+    # If there is one separator, it must be and entry point string.
+    # Check if the left hand side is a matching pattern
+    if ENTRY_POINT_STRING_SEPARATOR in identifier:
+        group_pattern, _ = identifier.split(ENTRY_POINT_STRING_SEPARATOR)
+        if not any(
+            _match_wildcard(string=group_name, pattern=group_pattern)
+            for group_name in entry_point_group_to_module_path_map
+        ):
+            raise ValueError(
+                common_error_msg + "Group name pattern '{}' does not match any of the AiiDA entry point group names.".
+                format(group_pattern)
+            )
+        # The group name pattern matches, and there are no further
+        # entry point string separators in the identifier, hence it is
+        # a valid pattern.
+        return
+    # The separator might be swallowed in a wildcard
+    if '*' in identifier:
+        group_part, _ = identifier.split('*', 1)
+        if any(group_name.startswith(group_part) for group_name in entry_point_group_to_module_path_map):
+            return
```

Comment:
is this block for identifiers like `aiida.*` and `aiida.calculations*`?

Reply:
Yes

Reply:
The reason for this one is that the `:` could be swallowed by the wildcard. For example, ``aiida.calc*2foo!!`` is, technically, valid. Without this part, it would be treated as a Python name and fail due to the invalid characters after the wildcard.

Reply:
Oh no... entry point names can contain the literal wildcard character 🤦‍♂